### DISCUSSION TITLE
Disable snap-by-dragging via terminal command

### DISCUSSION THREAD

**Discussion by reefdog** [rxhanson/Rectangle#718] [Q&A] [answered]
Hello @rxhanson, this is a wonderful piece of software and we're lucky to have it. Thank you!

I prefer to disable "Snap windows by dragging" entirely, which there is a UI checkbox for, but I'm having trouble finding the appropriate `defaults write` incantation. I think it's possible this is doable by combining [all of these](https://github.com/rxhanson/Rectangle/blob/master/TerminalCommands.md#ignore-specific-drag-to-snap-areas), but (1) I'm not sure how to do that (I'm bit-field-dumb), and (2) it'd be keen if there was a single attribute that would govern that.

I assume there is one since the checkbox exists, so would you be able to document what the correct command is?

Thank you much!

**Comment by rxhanson** [comment, marked as answer]
defaults write com.knollsoft.Rectangle windowSnapping -int 2

(0 or 1 will keep it enabled)

It's kind of unusual to do this via terminal command, since you can just copy over the plist in the event you want to fully automate Rectangle initialization, but no worries if you just needed that. https://github.com/rxhanson/Rectangle#preferences-storage

**Comment by reefdog** [comment]
Superb, thanks! I missed that preference file location, appreciate it. I was looking in `~/Library/Application Support/` for some reason. 🤦🏼

I like the `defaults write` approach because I use [dotfiles](https://github.com/reefdog/dotfiles/) to quickly setup new machines. The [`macos/set-defaults.sh`](https://github.com/reefdog/dotfiles/blob/main/macos/set-defaults.sh) file handles all the good stuff.
